portal/assets/js/global.js

```diff
@@ -68,20 +68,4 @@ document.addEventListener('DOMContentLoaded', function() {
     addMaildrop.parentNode.insertBefore(inputDropClone, addMaildrop);
   });
 
-
-  // Get user's infos
-  var r = new XMLHttpRequest();
-  r.open("GET", "/ynhpanel.json", true);
-  r.onreadystatechange = function () {
-    // Die if error
-    if (r.readyState != 4 || r.status != 200) return;
-    // Response is JSON
-    response = JSON.parse(r.responseText);
-
-    // load additional theme's script
-    loadScript("/yunohost/sso/assets/themes/"+ response.theme +"/js/ynhpanel.js");
```

Comment:
I'd also rather not have themes load js code. CSS in enough IMHO.

Reply:
Ah yes, in fact I don't know about this because I didn't do so much theming in my life. If you think JS ain't needed then that's fine with me ?

Reply:
No, not really. Most stuff can also be done using CSS.

Reply:
Personally, I think it would really good to have the possibility to run js in themes. If you want to do some real fun things, you will need it.

Reply:
Hmyea naively I too think that brings more flexibility ...

Reply:
But is a security hole. 
(Considering it might be possible in the future to 1-click-install random themes)

Reply:
Kind of yes, but I don't think we would allow to install themes which did not have some sort of reviewing from the community like we do with the community app list... So, yes, can be seen as a security hole, but not really much a bigger hole than installing community apps in the first place (which run bash scripts as root) :P

Reply:
I guess but that's why there would be official and non official themes (like for app) and a warning if you want to install external themes, no?
You have to trust someone at some point, when you install something.

Reply:
IMHO not letting js is keeping the power to do nice things only in the hands of YNH devs and to me, one cool thing of FOSS is when they make it easy for more people to do power stuff.
After, I don't know the whole extent of security issues, I just know that the first theme I made for YNH needed javascript to be achievable ;)

Reply:
Okay, I'm outvoted I guess :grin: 

Please take into account my comment to the other location where theme js is loaded, regarding the 404 that will occur with themes that have no js.

Reply:
I think it really boils down to what we allow in terms of "installing custom themes". For now I think we should do the following : 
- ship a few "official" themes
- allow people to easily add their own theme / tweakins, e.g. copying one of the default themes and tweaking the css and so on
- if people want to share theme with each other, they should wget and extract some zip archive (and of course realize that they are running other people's code on their machine ...) so still doable, but that requires more than 1-click.

In the long term : 
- We can allow people to change theme from the webadmin
- Allow people to import a new theme somehow (idk, maybe entering an url to a zip archive) and have a warning explaining that they should be careful ... Maybe we can have a repo with themes, and people would be able to submit PR to this repo to add their own theme to be reviewed by the community, idk. We have time to think about this.

But the bottom line is that we shall indeed avoid 1-click install where people ~~accept candies from strangers~~install themes from random third-parties without being aware of possible implications

Reply:
> Please take into account my comment to the other location where theme js is loaded, regarding the 404 that will occur with themes that have no js.

Yep, I also saw this issue when coding, but I didn't know a simple way to solve it. Checking that a file exist isn't so straightforward with js, and in the end I decided that it's not such a big deal if it just doesn't file the file. It's the same for styles, it just warn a 404 but doesn't really create an issue I think.
Another way would be to specify in the doc that it's good practice to have those files, even if they are empty.
But all that is not so optimized, I agree it would be cool to be able to make the request only if the file exists.

Reply:
> > Please take into account my comment to the other location where theme js is loaded, regarding the 404 that will occur with themes that have no js.
> 
> Yep, I also saw this issue when coding, but I didn't know a simple way to solve it. Checking that a file exist isn't so straightforward with js, and in the end I decided that it's not such a big deal if it just doesn't file the file. It's the same for styles, it just warn a 404 but doesn't really create an issue I think.
> Another way would be to specify in the doc that it's good practice to have those files, even if they are empty.
> But all that is not so optimized, I agree it would be cool to be able to make the request only if the file exists.

Yes, I'd try checking for the file in lua and printing some boolean value corresponding to this check into the js code, which in turn would stop the `script`-tag to be added when no js file was found.

Reply:
> But the bottom line is that we shall indeed avoid 1-click install where people ~~accept candies from strangers~~install themes from random third-parties without being aware of possible implications

:grin: 
If no 1-click install was available, the user can be blamed enough for putting malicious code into the application.

Reply:
@alexAubin I agree with all your points, I think the admin theme install part should be designed like the one for apps (which should/will be changed a bit I guess yunohost/issues#1322).
Official themes, easy to install in one click, non official with warnings, and make it easy to customize (duplicate, modify) themes.
I think a repo with themes is a cool idea.

Reply:
A repo with themes would be awesome!

It would be even more awesome if it was a submodule of SSOwat but idk if you're up for the hassle this brings :)